### DISCUSSION TITLE
Code Review Groups

### DISCUSSION THREAD

**Discussion by stoeckmann** [stoeckmann/letsreview#3] [Announcements]
You can add your own code review group here. Please add a short description about your group and how to contact you.

**Comment by stoeckmann** [comment]
C3H2-CTF Code Review Group. Contact: [Twitter](https://twitter.com/c3h2_ctf)
@khaleesicodes and @stoeckmann are part of it and used it for various references within the talk. Our focus is to improve our review skills by inspecting open source software and fixing bugs if we find some. Regular meetings, limited to CTF group members but always eager to get in touch with others!

**Comment by UweKrause** [comment]


**Reply by svenlampe** [reply]
php hab ich schon ewig nicht mehr gemacht...2004? Ich bin sicher, das ist hilfreich. Wenn es zeitlich passt, wär ich dabei.

**Reply by danto7** [reply]
Ich konnte leider gestern nicht dabei sein, aber an Nextcloud hättte ich auch interesse. Meine letztes PHP Projekt ist zwar auch schon 3 Jahre her aber ich bin gespannt das wieder ein wenig aufzufrischen.

**Reply by danto7** [reply]
Donnerstag passt mir nächste Woche nicht. Die anderen Tage passen aber für mich. Uhrzeiten sind alle fein für mich.

**Reply by UweKrause** [reply]
Diese Gruppe ist nicht zustande gekommen.
Ich habe versucht diesen Diskussionsfaden aufzuräumen, kann aber nicht alles löschen.

**Comment by sagischwarz** [comment]
Hi!

I would love to start a code reading group with the focus on the Rust programming language. I haven't done more than some tutorials and play projects, but I'm very interested in learning about the language, especially as a safer replacement for C/C++ and I think reading existing code can help a lot with that. It would also help me to get in contact with other interested beings and maybe find cool projects to contribute to. You can reach me here or on Twitter (@sagi_schwarz).

=)

**Reply by svenlampe** [reply]
That sounds interesting. I have no experience with Rust, but it's on my list ;-)

**Reply by Maaxxs** [reply]
I'm very interested in a rust focused code reviewing group, too :)

**Reply by kpcyrd** [reply]
I'm interested in a rust group too, do you have any suggestions for projects to review?

**Reply by sagischwarz** [reply]
Oh, some people are actually interested! :smile:

My idea was to start with some simple everyday tools. There is a repo that [re-implements the GNU core utils](https://github.com/uutils/coreutils) which I find quite interesting. I also found a [replacement for `ls`](https://github.com/ogham/exa), [one for `find`](https://github.com/sharkdp/fd) and [one for `cat`](https://github.com/sharkdp/bat). But I'm also very open to suggestions. :slightly_smiling_face:

We'd also have to think about when to meet and, if we want to continue, how often. [It has been suggested](https://github.com/stoeckmann/letsreview/blob/main/groups.md) to think about the format of the group (open/closed) and the goal, but we can probably talk about that in the first or following sessions.

**Reply by kpcyrd** [reply]
I'm fine with both, open or closed. I also know one of the devs of the uutils-coreutils and they'd be very likely open to patches.

**Reply by Maaxxs** [reply]
I'd prefer a closed/private group; at least in the beginning. 
I also wanted to look at [rebuilderd](https://github.com/kpcyrd/rebuilderd) for a couple of weeks now, but never really made time for it.  Not sure if this is a good/not so good idea as you, kpcyrd, wrote it afaikt :)

**Reply by sagischwarz** [reply]
I tend to agree that it is a good idea to start with a closed group.

As for the project, I think that it could become quite unbalanced and end up as a lecture if we review code written by one of the members of the group, at least in the beginning. Knowing that the maintainers of the coreutils are open for contributions is a huge plus, so I'd suggest to start with that.

What about a time for the first meeting? I'd prefer Tuesday or Thursday evening and I usually don't have time on Wednesday. I'm also often busy on weekends.

**Reply by stoeckmann** [reply]
Thanks for highlighting the fact that I did not talk about the advantages and disadvantages of these different "roles" in a group. I'll add some notes into the repository soon!

**Reply by Maaxxs** [reply]
Tuesday or Thursday evening is fine. 

Usually, the evenings during the week always work for me. Saturday or Sunday would be fine, too.

**Reply by miallo** [reply]
I am interested as well, but I also don't have experience with Rust. I am free most of the evenings, but I would also prefer weekdays

**Reply by stoeckmann** [reply]
Added information about roles and what to consider if the author is part of the group [here](https://github.com/stoeckmann/letsreview/blob/main/roles.md). Feel free to extend or ask if something is unclear through issues or pull requests!

**Reply by sagischwarz** [reply]
@miallo That's not a problem, I am also not very experienced. You are very welcome! If you would like to prepare yourself a bit, the [Rust Book](https://doc.rust-lang.org/book/) is very good and there is also [Rust By Example](https://doc.rust-lang.org/rust-by-example/). I also read a nice short introduction called [A half-hour to learn Rust](https://fasterthanli.me/articles/a-half-hour-to-learn-rust).

What do you all think about next Tuesday at 20:00 (UTC+1) for our first meeting? I would take care of the preparation.

**Reply by kpcyrd** [reply]
Tuesday works for me too :+1:

**Reply by Maaxxs** [reply]
Tuesday, 18.01.2022 at 20:00 UTC+1 it is 👍

**Reply by sagischwarz** [reply]
Great! I think we could use a Jitsi (https://jitsi.hamburg.ccc.de/RustCodeReadingGroup) for the meeting. Or does anyone have a better alternative?

**Reply by Maaxxs** [reply]
nope, sounds great 👍

**Reply by sagischwarz** [reply]
I did some serious bike shedding and created a GitHub organization for our group: https://github.com/TheBorrowCheckers. There is currently an almost empty public repo with basic information and a private one with orga stuff. I have added all of you as members. The name and everything else is, of course, open for discussion.

**Reply by stoeckmann** [reply]
Let us know when the setup is final so we can also add a link into our README. :)

**Reply by kpcyrd** [reply]
Should we create a separate github discussion thread? I'd then like to unsubscribe from the main thread. :)

**Reply by sagischwarz** [reply]
Sure. We could start new discussions in the orga repo of the org: https://github.com/TheBorrowCheckers/orga/discussions =)

**Reply by maz3max** [reply]
Can I join too?

**Reply by sagischwarz** [reply]
Sure, feel free to join today :)

**Reply by maz3max** [reply]
It's on the Jitsi at 20 today, right? I cannot view https://github.com/TheBorrowCheckers/orga/discussions yet.

**Reply by sagischwarz** [reply]
Yes, just join the Jitsi (https://jitsi.hamburg.ccc.de/RustCodeReadingGroup) at 20:00!

**Reply by sagischwarz** [reply]
> Let us know when the setup is final so we can also add a link into our README. :)

We had our first meeting. It was wonderful and we plan to continue on a regular basis. If @stoeckmann would like to add our group to your README, feel free to do so and link to our public about repo: https://github.com/TheBorrowCheckers/about.

**Reply by stoeckmann** [reply]
Great to hear, I have extended the Readme!

**Comment by NWuensche** [comment]
Ich mach hier mal noch einen JVM-Thread auf. Ich mach viel in meiner Freizeit mit Java/Kotlin, und hätte Lust drauf, Java/Kotlin/... Code zu reviewen. Hat da jemand Lust drauf?

**Reply by FridayToSundayFun** [reply]
Moin, bin etwas spät zur Party. Ich arbeite beruflich mit Java und wollte mir in nächster Zeit mal [Spring](https://github.com/spring-projects/spring-framework) anschauen. Wäre aber auch für [OpenJdK](https://github.com/openjdk/jdk) zu haben.

**Reply by NWuensche** [reply]
Das klingt gut. Ich hatte selber auch schon an Spring gedacht, und denke dass die OpenJDK vielleicht doch etwas zu hart ist ^^. Wann hättest du denn Zeit?

**Reply by FridayToSundayFun** [reply]
Generell würde ich gerne Dienstag anstreben, aber wie es immer so ist geht das nächste Woche leider nicht. Deswegen schlag ich einfach frech nächste Woche Mittwoch( 26.01) irgendwann zwischen 16-20 Uhr vor. Ich nehme auch gegen Vorschläge :P

**Reply by NWuensche** [reply]
Dann würde ich Mittwoch den 26.01 ab 17 Uhr, unter https://jitsi.hamburg.ccc.de/JavaCodeReadingGroup vorschlagen.

**Reply by FridayToSundayFun** [reply]
Ich freue mich, dann bis dahin.

**Reply by NWuensche** [reply]
Nächstes Treffen ist DIenstag, der 01.02, wieder um 17 Uhr.

**Reply by NWuensche** [reply]
Es hat sich eine Java Gruppe gegründet die gerne verlinkt werden kann: https://github.com/LetsReadJava/About

**Reply by stoeckmann** [reply]
Das Repository ist offenbar nicht öffentlich. Zumindest kann ich nicht darauf zugreifen (Organisation natürlich schon).

**Comment by Alienmaster** [comment]
Ich suche eine Gruppe um C++, Python oder Java Projekte zu reviewen. Die meisten Erfahrungen habe ich mit Python (diverse kleinere Projekte) und C++ (Arduino). Java ist schon ein paar Jahre her.

**Reply by UweKrause** [reply]
Python wäre natürlich auch cool!
Hast Du schon Ideen, in welche Richtung Du gehen möchtest?

**Reply by l-huel** [reply]
Hätte auf jeden Fall auch Interesse!

**Reply by Alienmaster** [reply]
Meine Idee wäre es sich Bibliotheken anzuschauen wie zB [redis-py](https://github.com/andymccurdy/redis-py), [opencv](https://github.com/opencv/) oder eine Standardbibliothek wie [csv](https://github.com/python/cpython/blob/3.10/Lib/csv.py).

**Reply by jonasha96** [reply]
Hallo :)
Ich hätte auch Interesse an einer Gruppe. Wollen wir uns nächste Woche mal treffen?

**Reply by Alienmaster** [reply]
Wie wäre es nächsten Donnerstag, 20.1. 20Uhr?

**Reply by jonasha96** [reply]
Ich hätte Zeit am Donnerstag um 20 Uhr.
Wollen wir uns dann auch über Jitsi treffen, wie die anderen Gruppen?

**Reply by odo2063** [reply]
Bei Python würde ich mich anschließen.

**Reply by UweKrause** [reply]
Wenn sich in der Web-Gruppe nichts ergibt, würde ich gerne bei euch mit reinschauen.
Do 20:00 ist super.

**Reply by Alienmaster** [reply]
Hier ein Jitsi Raum: https://jitsi.hamburg.ccc.de/DeadSnakes

**Reply by m0lenaar** [reply]
Hey :)
Ich hätte auch Interesse an einer Python-Gruppe. Do 20 Uhr klingt gut!

**Reply by Alienmaster** [reply]
Es hat sich eine Python Gruppe gegründet die gerne verlinkt werden kann: https://github.com/organizations/ThanatosisSerpenti

**Reply by stoeckmann** [reply]
Added. :)

**Comment by Maaxxs** [comment]
As far as I can see, no group covers C yet. Is anyone interested to review some C code? 
contact info: email in my github profile or via [twitter](https://twitter.com/Max_Kunzelmann)
Gibt es auch interessierte für eine Gruppe um C code zu reviewen/lesen?

**Reply by BenBE** [reply]
Might depend a bit on the project, but sure.

**Reply by Alienmaster** [reply]
Hast du Vorschläge für Projekte?

**Reply by Maaxxs** [reply]
- gnu-netcat and/or bsd-netcat.
- (open)ssh
- other command line utilities

open for suggestions

**Reply by Alienmaster** [reply]
- [coreutils](https://github.com/coreutils/coreutils/tree/master/src)
- [sox](https://github.com/chirlu/sox)
- [nano](https://git.savannah.gnu.org/cgit/nano.git/tree/)

for me it suits best in the evening during the week

**Reply by BenBE** [reply]
Würde vorschlagen: util-linux, lsof, iotop, iftop, ip, ethtool, isc-dhcp (client/server), openssl, …

**Reply by Maaxxs** [reply]
ok, also an Projekten mangelt es schon mal nicht. :D 
wie schauts mit dem ersten Termin aus: wie passt euch 
Mittwoch, 19.01.2022 um 20:00 utc+1 ?

**Reply by Alienmaster** [reply]
Passt mir. Wo treffen wir uns?

**Reply by BenBE** [reply]
Würd analog der Rust-Gruppe [Jitsi](https://jitsi.hamburg.ccc.de/CCodeReadingGroup) vorschlagen.

**Reply by BenBE** [reply]
FYI, am Mittwoch haben wir - als Casual C Code Reviewers - bei unserem ersten Treffen in `iproute2` reingeschaut und werden dies an gleicher Stelle diesen Mittwoch, ab 19:00 UTC+0100 an gleicher Stelle fortsetzen.

Organisation gibt es auch bereits, aber bisher sind wir da noch nicht so weit, dass wir da was public gesetzt haben. Sobald es an der Front etwas Neues gibt, erfolgt hier natürlich die Meldung; bis dahin bei Interesse hier kurz mit Reaktion auf den Post interagieren und zur richtigen Zeit™ auf dem Jitsi vorbeischauen.

**Reply by Maaxxs** [reply]
Unsere [about page](https://github.com/CasualCCodeReviewers/about) kann nun auch gern verlinkt werden.

**Comment by svenlampe** [comment]
Thanks,

that are some good starting points. I don‘t have time in the next week but in 2 weeks I have time for a meeting. I don‘t have time on sundays (p&p round) but during the week evenings it’s possible to meet. It depends on the week because I participate in different committees (Stadtelternrat, Schulausschuss, usw. )
I try to participate in code review meetings but I might not always be there. 

Best regards

Sven 

Beste Grüße 

Sven

**Comment by UweKrause** [comment]


**Reply by kpcyrd** [reply]
Should we track this in a different [discussion](https://github.com/stoeckmann/letsreview/discussions)? :) We're currently keeping everything in one threaded-thread that's becoming hard to navigate. /o\

**Reply by stoeckmann** [reply]
Sounds like a good idea. For end of next week. As far as I follow this discussion correctly, three groups will have had their initial meetings by then. Maybe we can add links to their newly formed github organizations or the groups create a new discussion thread, too.

I am fine with both. Feel free to use this repository as long as needed. :)
